doc/src/contracts/connection_router.md

```diff
@@ -0,0 +1,139 @@
+# Connection Router Contract
+
+The connection router contract is responsible for routing messages to a chain and controlling gateways associated with a chain that is connected to Axelar.
+
+## Interface
+
+```Rust
+pub enum ExecuteMsg {
+    /*
+     * Governance Methods
+     * All of the below messages can only be called by governance
+     */
+    // Registers a new chain with the router
+    RegisterChain {
+        chain: ChainName,
+        gateway_address: String,
+    },
+    // Changes the gateway address associated with a particular chain
+    UpgradeGateway {
+        chain: ChainName,
+        contract_address: String,
+    },
+
+    /*
+     * Router Admin Methods
+     * All of the below messages can only be called by the router admin
+     */
+    // Freezes a chain, in the specified direction.
+    FreezeChain {
+        chain: ChainName,
+        direction: GatewayDirection,
+    },
+    // Unfreezes a chain, in the specified direction.
+    UnfreezeChain {
+        chain: ChainName,
+        direction: GatewayDirection,
+    },
+
+    /*
+     * Gateway Messages
+     * The below messages can only be called by registered gateways
+     */
+    // Routes a message to all outgoing gateways registered to the destination domain.
+    // Called by an incoming gateway
+    RouteMessages(Vec<Message>),
+}
+
+#[derive(QueryResponses)]
+pub enum QueryMsg {}
+```
+
+## Events
```

Comment:
I don't think we need this section. This is a bit too low level for this doc, and these events are not mentioned outside of this section.